### PR TITLE
v.distance: relax parser rules

### PR DESCRIPTION
**Pull request by ninsbl**
This is a basic quick-fix for #2126

Tests should be added...

### CODE REVIEW

File: vector/v.distance/main.c [hunk 1]
```diff
@@ -266,7 +266,8 @@ int main(int argc, char *argv[])
     opt.to_field->guidependency = G_store(opt.to_column->key);
 
     G_option_required(opt.out, opt.table, flag.print, opt.column, NULL);
-    G_option_exclusive(opt.out, opt.table, flag.print, NULL);
+    G_option_exclusive(opt.out, flag.print, NULL);
+    G_option_exclusive(opt.table, flag.print, NULL);
```
Comment (wenzeslaus) [inline vector/v.distance/main.c:270]: Wasn't the idea in the original #601 code _G_option_excludes_ (when first is present, none of the other options should be present)?

```c
G_option_excludes(flag.print, opt.out, opt.table, NULL);
```

**Review by wenzeslaus — APPROVED** [review]
Thanks for quickly addressing this.

The original PR #601 changes the rules, but does not comment on in the the description nor the commit message and with all these options it is hard for me to get a full picture fast enough.

However, the current code here follows the manual and fits with the state before #601 where `opt.table` actually requires `opt.out`:

```c
G_option_requires_all(opt.table, opt.out, opt.upload, NULL);
```

I think it can be rewritten using one function call, but that's optional. Perhaps for a later PR with test.

File: vector/v.distance/main.c (hunk 1 above)
Comment (ninsbl) [reply vector/v.distance/main.c]: Yes, yu are right. Implemented in 3f7f04f

**Comment by ninsbl** [comment]
Unfortunately, now I get a `segmentation fault` error...

**Comment by ninsbl** [comment]
OK. Now I ran it through `gdb`:

```
Starting program: /opt/src/grass_ninsbl/dist.x86_64-pc-linux-gnu/bin/v.distance --overwrite --verbose from=railroads@PERMANENT to=railroads@PERMANENT output=t dmin=0.1 upload=dist table=t
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
WARNING: Vector map <t> already exists and will be overwritten
Using native format
Finding nearest features...
 100%

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7f2518c in db_start_driver (name=0x0) at start.c:97
97	    if (name[0] == '\0') {
```


So, the culprit seems:
https://github.com/OSGeo/grass/pull/1265
in particular:
https://github.com/OSGeo/grass/blob/882597123599767abd02f1486724ea02914372f1/lib/db/dbmi_client/start.c#L97

Replacing `name[0]` with `name` makes `v.distance` produce output again. However, it was changed on purpose, so, I wonder if a different solution would be needed. Any idea @nilason ?
Changing the line to:
```
    if (name == NULL) {
```
both removes the compiler warning and makes the module run again...
Should I add the latter approach to the PR?

**Comment by wenzeslaus** [comment]
Maybe `if(name == NULL || name[0] == '\0') {` is needed? Check whether or not the pointer is null point and then check that the provided string is actually filled. That would be equivalent with Python `if not name` which works like `if name is None or len(name) == 0`. The intention in the original code seems to be to check if the string is empty. Well, that's at least the understanding of the compiler producing the warning which says "did you mean to dereference the pointer?" However, according to the documentation, NULL is a valid value of _name_, so that needs to be checked and that's probably the actual intention there (and `\0` converts to NULL (`(char *)'\0'` so the original code worked).

So `name == NULL` would probably do, but covering also the "is empty" case (`name[0] == '\0'`) seems appropriate.

Since this is in the library (unrelated to v.distance parser rules), I'm afraid another PR is needed.

For the record, I was quite confused before I understood the relation of #1265 and f0261d556c577048f3e30546c0a324e6f44ad3d3. Let's not do that again :-)

**Comment by ninsbl** [comment]
OK. The library fix is now implemented with `if(name == NULL || name[0] == '\0')` in #2131 
Is it OK to backport this to 8.0 (or should we wait for 8.0.1)?

**Pull request merged**